Cython/Utility/ImportExport.c

```diff
@@ -238,6 +238,34 @@ static PyObject* __Pyx_ImportFrom(PyObject* module, PyObject* name); /*proto*/
 static PyObject* __Pyx_ImportFrom(PyObject* module, PyObject* name) {
     PyObject* value = __Pyx_PyObject_GetAttrStr(module, name);
     if (unlikely(!value) && PyErr_ExceptionMatches(PyExc_AttributeError)) {
+        /* name may refer to a (sub-)module which has not finished initialization
+         * yet, and may not be assigned as an attribute to its parent, so try
+         * finding it by full name */
+        const char* module_name_str = 0;
+        PyObject* module_name = 0;
+        PyObject* dot = 0;
+        PyObject* module_dot = 0;
+        PyObject* full_name = 0;
+        PyErr_Clear();
+        module_name_str = PyModule_GetName(module);
+        if (unlikely(!module_name_str)) { goto modbad; }
+        module_name = PyUnicode_FromString(module_name_str);
+        if (unlikely(!module_name)) { goto modbad; }
+        dot = PyUnicode_FromString(".");
+        if (unlikely(!dot)) { goto modbad; }
+        module_dot = PyUnicode_Concat(module_name, dot);
```

Comment:
```suggestion
        module_dot = PyUnicode_Concat(module_name, PYUNICODE("."));
```

This is a shortcut Cython supports for utility code for Unicode constants - it gets substituted when the source file is generated and just avoids you having to manage them manually. You then don't need to manage `PyObject *dot` (so can remove some of your lines above)

Reply:
Thanks, committed changes!